extractor/filesystem/internal/cachedregexp/regex.go

```diff
@@ -0,0 +1,22 @@
+// Package cachedregexp provides a cached MustCompile alternative to regexp
+package cachedregexp
+
+import (
+	"regexp"
+	"sync"
+)
+
+var cache sync.Map
```

Comment:
Once concern that came up is that creating a lot of map structures could be expensive memory-wise for envs with resource constraints. If we just want to make sure that a regexp isn't compiled several times can't we just store the regexes as global vars? That's what we do in SCALIBR currently, e.g. https://github.com/google/osv-scalibr/blob/main/converter/converter.go#L42

Reply:
Removed the entire map for now.